File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Repositories/RepositorioTurmaJson.cs
using SistemaAcademico.Models;
using SistemaAcademico.Exceptions;

namespace SistemaAcademico.Repositories;

// Equivalente à classe Java: RepositorioTurmaJson implements RepositorioTurma
// A geração de JSON é manual (string building), idêntica à implementação Java original.
// Não usamos System.Text.Json para manter fidelidade ao formato e estrutura originais.
public class RepositorioTurmaJson : IRepositorioTurma
{
    private string arquivo = "turmas.json";

    public RepositorioTurmaJson() { }

    public RepositorioTurmaJson(string arquivo)
    {
        this.arquivo = arquivo;
    }

    public void SalvarTodas(List<Turma> turmas)
    {
        try
        {
            using StreamWriter writer = new StreamWriter(arquivo);
            writer.WriteLine("{");
            writer.WriteLine("  \"turmas\": [");
            for (int i = 0; i < turmas.Count; i++)
            {
                Turma turma = turmas[i];
                writer.WriteLine("    {");
                writer.WriteLine("      \"codigo\": \"" + turma.Codigo + "\",");
                writer.WriteLine("      \"titulo\": \"" + turma.Titulo + "\",");
                writer.WriteLine("      \"avaliacoes\": [");
                List<Avaliacao> avaliacoes = turma.GetAvaliacoes();
                for (int j = 0; j < avaliacoes.Count; j++)
                {
                    Avaliacao avaliacao = avaliacoes[j];
                    writer.WriteLine("        {");
                    writer.WriteLine("          \"tipo\": \"" + avaliacao.GetType().Name + "\",");
                    writer.WriteLine("          \"valor\": " + avaliacao.Valor + ",");
                    writer.WriteLine("          \"peso\": " + avaliacao.Peso);
                    if (j < avaliacoes.Count - 1) writer.WriteLine("        },");
                    else writer.WriteLine("        }");
                }
                writer.WriteLine("      ]");
                if (i < turmas.Count - 1) writer.WriteLine("    },");
                else writer.WriteLine("    }");
            }
            writer.WriteLine("  ]");
            writer.WriteLine("}");
        }
        catch (IOException e)
        {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo JSON: " + e.Message);
        }
    }
}

File: academic-system-java/src/main/java/org/example/repository/RepositorioTurmaJson.java
package org.example.repository;

import org.example.model.Turma;
import org.example.model.Avaliacao;
import java.util.List;
import java.io.FileWriter;
import java.io.PrintWriter;
import java.io.IOException;
import org.example.exception.ExcecaoSistemaAcademico;

public class RepositorioTurmaJson implements RepositorioTurma {
    
    private String arquivo = "turmas.json";

    public RepositorioTurmaJson() {}
    public RepositorioTurmaJson(String arquivo) { this.arquivo = arquivo; }

    @Override
    public void salvarTodas(List<Turma> turmas) {
        try (PrintWriter writer = new PrintWriter(new FileWriter(arquivo))) {
            writer.println("{");
            writer.println("  \"turmas\": [");
            for (int i = 0; i < turmas.size(); i++) {
                Turma turma = turmas.get(i);
                writer.println("    {");
                writer.println("      \"codigo\": \"" + turma.getCodigo() + "\",");
                writer.println("      \"titulo\": \"" + turma.getTitulo() + "\",");
                writer.println("      \"avaliacoes\": [");
                List<Avaliacao> avaliacoes = turma.getAvaliacoes();
                for (int j = 0; j < avaliacoes.size(); j++) {
                    Avaliacao avaliacao = avaliacoes.get(j);
                    writer.println("        {");
                    writer.println("          \"tipo\": \"" + avaliacao.getClass().getSimpleName() + "\",");
                    writer.println("          \"valor\": " + avaliacao.getValor() + ",");
                    writer.println("          \"peso\": " + avaliacao.getPeso());
                    writer.print("        }");
                    if (j < avaliacoes.size() - 1) writer.println(",");
                    else writer.println();
                }
                writer.println("      ]");
                writer.print("    }");
                if (i < turmas.size() - 1) writer.println(",");
                else writer.println();
            }
            writer.println("  ]");
            writer.println("}");
        } catch (IOException e) {
            throw new ExcecaoSistemaAcademico("Erro ao salvar dados no arquivo JSON: " + e.getMessage());
        }
    }
}
